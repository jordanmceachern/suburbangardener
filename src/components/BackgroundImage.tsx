import React from "react";

export interface BackgroundImageProps {
  className?: string;
  imageClassName?: string;
  height?: number;
  label?: string;
  src: string;
  width?: number;
  minWidth?: number;
  maxWidth?: number;
}

export function BackgroundImage({
  className,
  height,
  imageClassName,
  label,
  maxWidth,
  minWidth,
  src,
  width,
}: BackgroundImageProps) {
  const customContainerHeight = {
    height: height ? `${height}px` : undefined,
  };
  const customImageWidth = {
    maxWidth: maxWidth ? `${maxWidth}px` : undefined,
    minWidth: minWidth ? `${minWidth}px` : undefined,
    width: width ? `${width}px` : undefined,
  };
  return (
    <div
      className={`h-auto w-full flex justify-center${className ? " " + className : ""}`}
      style={customContainerHeight}
    >
      <div
        aria-label={label}
        className={`bg-contain bg-center bg-no-repeat w-full h-full dark:bg-neutral-300 dark:rounded-lg dark:shadow-lg${imageClassName ? " " + imageClassName : ""} rounded-lg`}
        style={{ backgroundImage: `url(${src})`, ...customImageWidth }}
      />
    </div>
  );
}

export default BackgroundImage;
