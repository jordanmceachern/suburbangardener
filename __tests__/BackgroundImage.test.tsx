import { render } from "@testing-library/react";
import BackgroundImage from "@/components/BackgroundImage";

describe("BackgroundImage Component", () => {
  const defaultProps = {
    src: "/test-image.jpg",
    label: "Test Background",
  };

  it("should match DOM snapshot", () => {
    const { container } = render(<BackgroundImage {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for background styling", () => {
    const { container } = render(
      <BackgroundImage {...defaultProps} className="custom-bg-class" />
    );

    // Check for background-related classes
    const hasBackgroundClasses =
      container.innerHTML.includes("bg-") ||
      container.innerHTML.includes("background") ||
      container.innerHTML.includes("backdrop") ||
      container.innerHTML.includes("custom-bg-class");

    expect(hasBackgroundClasses).toBe(true);
  });
});
