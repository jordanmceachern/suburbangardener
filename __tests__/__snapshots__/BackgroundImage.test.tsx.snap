// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`BackgroundImage Component should match DOM snapshot 1`] = `
<div
  class="h-auto w-full flex justify-center"
>
  <div
    aria-label="Test Background"
    class="bg-contain bg-center bg-no-repeat w-full h-full  rounded-lg"
    style="background-image: url("/test-image.jpg");"
  />
</div>
`;
